src/components/Dropdown/Dropdown.tsx

```diff
@@ -222,48 +229,93 @@ export default class Dropdown extends AutoControlledComponent<
             getRootProps,
             getToggleButtonProps,
             isOpen,
+            toggleMenu,
             highlightedIndex,
             selectItemAtIndex,
           }) => {
+            const accessibilityRootProps = getRootProps(
+              { refKey: 'innerRef' },
+              { suppressRefError: true },
+            )
+            const { innerRef, ...accessibilityRootPropsRest } = accessibilityRootProps
             return (
-              <div
-                className={classes.containerDiv}
-                onClick={this.handleContainerClick.bind(this, isOpen)}
-              >
-                {multiple && this.renderSelectedItems(styles)}
-                {search &&
-                  this.renderSearchInput(
-                    getRootProps,
-                    getInputProps,
+              <Ref innerRef={innerRef}>
+                <div
+                  className={classes.container}
+                  onClick={multiple ? this.handleContainerClick.bind(this, isOpen) : undefined}
+                >
+                  {multiple && this.renderSelectedItems(styles)}
+                  {search
+                    ? this.renderSearchInput(
+                        accessibilityRootPropsRest,
+                        getInputProps,
+                        highlightedIndex,
+                        selectItemAtIndex,
+                        variables,
+                      )
+                    : this.renderTriggerButton(styles, getToggleButtonProps)}
+                  {toggleButton && this.renderToggleButton(getToggleButtonProps, styles, isOpen)}
+                  {this.renderItemsList(
+                    styles,
+                    variables,
+                    isOpen,
                     highlightedIndex,
+                    toggleMenu,
                     selectItemAtIndex,
+                    getMenuProps,
+                    getItemProps,
+                    getInputProps,
                   )}
-                {toggleButton && this.renderToggleButton(getToggleButtonProps, styles, isOpen)}
-                {this.renderItemsList(
-                  styles,
-                  variables,
-                  getMenuProps,
-                  getItemProps,
-                  isOpen,
-                  highlightedIndex,
-                )}
-              </div>
+                </div>
+              </Ref>
             )
           }}
         </Downshift>
       </ElementType>
     )
   }
 
+  private renderTriggerButton(
+    styles: ComponentSlotStylesInput,
+    getToggleButtonProps: (options?: GetToggleButtonPropsOptions) => any,
+  ): JSX.Element {
+    const { placeholder, itemToString, multiple } = this.props
+    const { value } = this.state
+    const content = multiple ? placeholder : value ? itemToString(value) : placeholder
+    return (
+      <Ref
+        innerRef={buttonNode => {
+          this.buttonNode = buttonNode
+        }}
+      >
```

Comment:
Please use  `createRef()` API: https://reactjs.org/docs/refs-and-the-dom.html#creating-refs
This is more simple and effective.

Reply:
Then I will get a ref object and it will point to my Button component, that cannot be focused.
If I need to do more magic to access the DOM button, I would rather leave it like this.

Reply:
Why? It does the same thing as callback.

```diff
-private buttonNode: HTMLElement
+private buttonRef = React.createRef<HTMLElement>()
```

```diff
-      <Ref
-        innerRef={buttonNode => {
-          this.buttonNode = buttonNode
-        }}
-      >
+<Ref innerRef={this.buttonRef}>
```

```diff
-this.buttonNode.focus()
+this.buttonNode.current.focus()
```

This is more clear and simple.

Reply:
BTW, we a have a decision to name refs to as ***Ref**: https://github.com/stardust-ui/react/pull/459#discussion_r233852203

Reply:
I was using it as in the React docs, `ref={this.buttonRef}`. Will refactor, thanks.